File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/comment/uncomment/direct/Java/C/s967693306_commented.c
#include <stdio.h>
#include <string.h>

int main() {
    int n, m;
    scanf("%d %d", &n, &m);
    int required[10] = {0};
    for (int i = 0; i < 10; i++) required[i] = -1;

    for (int i = 0; i < m; i++) {
        int s, c;
        scanf("%d %d", &s, &c);
        if (required[s] != -1 && required[s] != c) {
            printf("-1\n");
            return 0;
        }
        required[s] = c;
    }

    for (int i = 0; i <= 999; i++) {
        char s_str[5];
        sprintf(s_str, "%d", i);
        int len = strlen(s_str);
        if (len != n) continue;

        int flag = 1;
        for (int j = 0; j < n; j++) {
            int digit = s_str[j] - '0';
            int pos = j + 1;
            if (required[pos] != -1) {
                if (digit != required[pos]) {
                    flag = 0;
                    break;
                }
            } else {
                if (j == 0) {
                    if (digit != 1) {
                        flag = 0;
                        break;
                    }
                } else {
                    if (digit != 0) {
                        flag = 0;
                        break;
                    }
                }
            }
        }

        if (flag) {
            printf("%d\n", i);
            return 0;
        }
    }

    printf("-1\n");
    return 0;
} // End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/Java/Go/s967693306_commented.go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	var n, m int
	fmt.Scanf("%d %d", &n, &m)
	constraints := make(map[int]int)
	conflict := false

	for i := 0; i < m; i++ {
		var s, c int
		fmt.Scanf("%d %d", &s, &c)
		if existing, ok := constraints[s]; ok {
			if existing != c {
				fmt.Println(-1)
				return
			}
		} else {
			constraints[s] = c
		}
	}

	found := false
	ans := ""
	for i := 0; i < 1000; i++ {
		s := strconv.Itoa(i)
		if len(s) != n {
			continue
		}
		flag := true
		for j := 0; j < n; j++ {
			currentDigit := int(s[j] - '0')
			pos := j + 1
			required, exists := constraints[pos]
			if exists {
				if currentDigit != required {
					flag = false
					break
				}
			} else {
				if j == 0 {
					if currentDigit != 布 {
						flag = false
						break
					}
				} else {
					if currentDigit != 0 {
						flag = false
						break
					}
				}
			}
		}
		if flag {
			ans = s
			found = true
			break
		}
	}

	if found {
		fmt.Println(ans)
	} else {
		fmt.Println(-1)
	}
}

// End of Code
